File: product.cpp
#include "Product.h"
#include <iostream>

Product::Product() : id(""), name(""), price(0), quantityInStock(0) {}

Product::Product(std::string id, std::string name, double price, int quantity)
    : id(id), name(name), price(price), quantityInStock(quantity) {}

void Product::setQuantity(int quantity) {
    quantityInStock = quantity;
}

void Product::setPrice(double price) {
    this->price = price;
}

std::string Product::getId() const {
    return id;
}

std::string Product::getName() const {
    return name;
}

double Product::getPrice() const {
    return price;
}

int Product::getQuantity() const {
    return quantityInStock;
}

void Product::display() const {
    std::cout << "Product ID: " << id << "\n"
              << "Name: " << name << "\n"
              << "Price: " << price << "\n"
              << "Stock: " << quantityInStock << "\n";
}

File: SupermarketUI/Models/Product.cs
using System;

namespace SupermarketUI.Models
{
    public class Product
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public float Price { get; set; }
        public int Quantity { get; set; }

        public bool IsAvailable => Quantity > 0;

        public override string ToString()
            => $"{Name} ({Id}) – ${Price} × {Quantity}";
    }
}